packages/cozy-stack-client/src/FileCollection.js

```diff
@@ -721,14 +721,39 @@ class FileCollection extends DocumentCollection {
     return formatBytes(parseInt(file.size), decimal)
   }
 
-  async downloadArchive(fileIds, notSecureFilename = 'files') {
-    const filename = slugify(notSecureFilename)
-    const href = await this.getArchiveLinkByIds(fileIds, filename)
+  /**
+   * Attributes used for create archive link by ids
+   *
+   * @typedef {object} ArchivePages
+   * @property {string} id - Id of the file
+   * @property {number} page - The page number
+   */
+
+  /**
+   * Download an archive of the files
+   *
+   * @param {string[]} fileIds - List of file ids
+   * @param {string} [notSecureFilename] - Name of the archive (default: 'files')
+   * @param {ArchivePages[]} [pages] - Array of objects, with `id` the file identifier, and `page` the page number (1 is the first page)
```

Comment:
The naming of `ArchivePages` and pages does not sound right here, as the archive could be non-pdf files, and thus without any concept of pages.